src/test/java/hudson/plugins/sshslaves/verifiers/TrustHostKeyActionTest.java

```diff
@@ -93,29 +85,33 @@ public void testSubmitNotAuthorised() throws Exception {
         
         final int port = findPort();
 
-        SshServer server = SshServer.setUpDefaultServer();
-        server.setPort(port);
-        server.setKeyPairProvider(new PEMGeneratorHostKeyProvider());
-        server.setUserAuthFactories(Arrays.asList((NamedFactory<UserAuth>)new UserAuthPassword.Factory()));
-        server.setCommandFactory(new CommandFactory() {
+        try {
+            Object server = newSshServer();
+            assertNotNull(server);
+            Class keyPairProviderClass = newKeyPairProviderClass();
+            Object provider = newProvider();
+            assertNotNull(provider);
+            Object factory = newFactory();
+            assertNotNull(factory);
+            Class commandFactoryClass = newCommandFactoryClass();
+            Object commandFactory = newCommandFactory(commandFactoryClass);
+            assertNotNull(commandFactory);
+            Class commandAuthenticatorClass = newCommandAuthenticatorClass();
+            Object authenticator = newAuthenticator(commandAuthenticatorClass);
+            assertNotNull(authenticator);
 
-            @Override
-            public Command createCommand(final String command) {
-                return new UnknownCommand(command);
-            }
-            
-        });
-        server.setPasswordAuthenticator(new PasswordAuthenticator() {
-            
-            @Override
-            public boolean authenticate(String username, String password, ServerSession session) {
-                return true;
-            }
-        });
+            invoke(server, "setPort", new Class[] {Integer.TYPE}, new Object[] {port});
+            invoke(server, "setKeyPairProvider", new Class[] {keyPairProviderClass}, new Object[] {provider});
+            invoke(server, "setUserAuthFactories", new Class[] {List.class}, new Object[] {Arrays.asList(factory)});
+            invoke(server, "setCommandFactory", new Class[] {commandFactoryClass}, new Object[] {commandFactory});
+            invoke(server, "setPasswordAuthenticator", new Class[] {commandAuthenticatorClass}, new Object[] {authenticator});
+
+            invoke(server, "start", null, null);
+        } catch (ClassNotFoundException | NoSuchMethodException | InvocationTargetException | IllegalAccessException | InstantiationException | IllegalArgumentException e) {
+            e.printStackTrace();
+            throw new AssertionError("Check sshd-core version");
```

Comment:
Why not just let these be thrown up?

Reply:
IMO, this message is more easy to understand than an InvocationTargetException (e.g.) and gives the root cause of a possible error. However, I agree with you about throwing the original exception to have the detailed error so I will change it into 
`throw new AssertionError("Check sshd-core version", e);`